compact/ecal/barrel_interlayers_9layers.xml

```diff
@@ -0,0 +1,192 @@
+<!-- SPDX-License-Identifier: LGPL-3.0-or-later -->
+<!-- Copyright (C) 2022 Whitney Armstrong, Chao Peng, Maria Zurek, Jihee Kim -->
```

Comment:
This file is identical to `barrel_interlayers.xml`, except for `EcalBarrelImagingLayers_nMax` which is 9 instead of 6. This feels like we should approach this differently (under the "don't repeat yourself" mantra), and this is an instructive example to start the discussion with.

In this case, we could separate out the definition of  `EcalBarrelImagingLayers_nMax`, and put it in a file `barrel_interlayers_6layers.xml` with just this:
```xml
<lccdd>
  <define>
     <constant name="EcalBarrelImagingLayers_nMax"   value="6"/>
  </define>
</lccdd>
```
and similar for the `barrel_interlayers_9layers.xml`.

The current file `barrel_interlayers.xml` would become something like `barrel_interlayers_impl.xml` and would not have any `EcalBarrelImagingLayers_nMax` definition.